docs/rfcs/0001-rfc-process.md

```diff
@@ -0,0 +1,64 @@
+---
+layout: page
+title: RFC process
+permalink: /rfc/0001-rfc-process
+---
+
+* Start date: 2020-07-07
+* RFC PR: [#7651](https://github.com/mozilla-mobile/android-components/pull/7651)
+
+## Summary
+
+Introducing a lightweight RFC ("request for comments") process for proposing and discussing "substantial" changes and for building consensus.
+
+## Motivation
+
+For many changes the existing workflow of opening and reviewing pull requests is fully sufficient.
+
+For larger, substantial changes (functionality, behavior, architecture), before writing any code, an RFC process may help with:
+
+* Publicly discussing a change proposal with other maintainers and consumers of components.
+* Gathering and integrating feedback into a proposal.
+* Documenting why specific changes and decisions were made.
+* Building consensus among the team before potentially writing a lot of code.
+
+## Guide-level explanation
+
+The high-level process of creating an RFC is:
+
+* Create an RFC document (like this one) using the template.
+* Open a pull request for the RFC document.
+* Ask for feedback on the pull request, via the [mailing list]() or in [chat](https://chat.mozilla.org/#/room/#android-components:mozilla.org).
+
```

Comment:
Personally, I would also like to see some sample code. Do you think that's reasonable requirement? It doesn't have to be working, it could be a sketch as part of the RFC PR or a link to a draft PR. It will make it much easier to understand the bigger picture. Of course, it has to be clear that this is not about the specific code, but the design / architecture.

Reply:
In addition to this RFC (once landed) I will write a documentation (I can share my draft) that includes a template that can be copied and pasted to start an RFC. Do you think that would be sufficient?

Reply:
Yes, sounds good. If there's a snippet part of the template I think that's sufficient.